### PR TITLE
enable v3 transaction on deploy

### PR DESCRIPTION
**Pull request by ksew1**
**Stack**:
- #2301
- #2300 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Review by Arcticae — APPROVED** [review]
LGTM

File: crates/sncast/src/helpers/error.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        indoc! {
+            r"
+            {} fee token is not supported for {} deployment.
+
+            Possible values:
+            +---------+----------+
+            | Version | FeeToken |
+            +---------+----------+
+            | v1      | eth      |
+            | v3      | strk     |
```
[the remark this answers is not in the document]
Comment (ksew1) [reply crates/sncast/src/helpers/error.rs:15]: Only lower case is. We can support upper case, but i don't know if it is worth to break out of convention, as in most of cli tool args are in lower case. Clap will give error like this:

```
error: invalid value 'V1' for '--version <VERSION>'
  [possible values: v1, v3]
```

File: crates/sncast/src/helpers/error.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#[must_use]
+pub fn token_not_supported_error_msg(fee_token: &str, deployment: &str) -> String {
+    format!(
+        indoc! {
+            r"
+            {} fee token is not supported for {} deployment.
+
+            Possible values:
+            +---------+----------+
+            | Version | FeeToken |
```
[the remark this answers is not in the document]
Comment (ksew1) [reply crates/sncast/src/helpers/error.rs:12]: My mistake. there should be a space between Fee and Token.

File: crates/sncast/src/helpers/error.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Arcticae) [reply crates/sncast/src/helpers/error.rs]: Agreed to fix case sensitivity in other PR

**Comment by ksew1** [comment]
✓ Commit merged in pull request [#2301](https://github.com/foundry-rs/starknet-foundry/pull/2301)

**Pull request closed without merging**